### PR TITLE
CDRIVER-4435 Add support for GCP service accounts in GCP KMS 

### PR DESCRIPTION
**Pull request by galon1**
## Summary:
1. upgrade libmongocrypt dependency to 1.6.0 
2. If the KMS provider is set up with an empty `gcp: {}`, create a `gcp_service_account_token` by sending a HTTP request to the GCP metadata server. Most of these changes are modeled after how azure tokens were added.  
3. Testing file for `service-gcp.c` which uses a mock GCP server to test the code handles different HTTP responses. This test uses the `fake_azure.py` mock server.
4. Integration test for GCP. Adds a new task group in `.evergreen/config.yml` to spin up a remote GCE instance using the scripts in `drivers-evergreen-tools` ([link to github](https://github.com/mongodb-labs/drivers-evergreen-tools/tree/master/.evergreen/csfle/gcpkms)). Here is a patch [build](https://spruce.mongodb.com/version/6362cf9ae3c331270cf4c53c/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) with the new GCP KMS tasks.

## Notes:
the files listed below were copied from this [PR ](https://github.com/mongodb/mongo-c-driver/pull/1124#event-7643575802) since they will be added late, but are needed for integration testing.
1. [build/evergreen_config_generator/taskgroups.py](https://github.com/mongodb/mongo-c-driver/compare/master...galon1:mongo-c-driver:CDRIVER-4435?expand=1#diff-974bdd422c8cf802e9d17ce4ab23440a2bcc19601adf9560e4f0024f8bd45c25)
2. [build/evergreen_config_lib/taskgroups.py](https://github.com/mongodb/mongo-c-driver/compare/master...galon1:mongo-c-driver:CDRIVER-4435?expand=1#diff-261f6a621f7139535e9f71f484eed10e0a7906818c8f30337a43c674460c6034)

Most recent patch build with the new tasks can be found here: https://spruce.mongodb.com/version/6363fa95d1fe076083e2114f/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

### CODE REVIEW

File: build/evergreen_config_lib/testgcpkms.py
Comment (vector-of-bool) [inline build/evergreen_config_lib/testgcpkms.py:45]: Trailing whitespace on this line (35) is causing the YAML output string to be garbled in `config.yml`.

File: build/evergreen_config_lib/testgcpkms.py
Comment (vector-of-bool) [inline build/evergreen_config_lib/testgcpkms.py:65]: Trailing whitespace after `popd` is garbling the generated string in `config.yml`

File: build/fake_azure.py
Comment (vector-of-bool) [inline build/fake_azure.py:88]: Aside: If we're using this file for multiple mock services, it may be useful to rename it and/or split it into multiple Bottle applications.

File: src/libmongoc/src/mongoc/mongoc-error.h
Comment (vector-of-bool) [inline src/libmongoc/src/mongoc/mongoc-error.h:146]: Maybe these should be merged since errors are already distinguished by the error domain.

File: src/libmongoc/tests/test-service-gcp.c
Comment (vector-of-bool) [inline src/libmongoc/tests/test-service-gcp.c:78]: This env var (and the test fixture) should be renamed since it is used for both Azure and GCP testing.

**Review by vector-of-bool — CHANGES_REQUESTED** [review]
Some changes requested, some about consolidating with Azure KMS work.

File: build/fake_azure.py
Comment (galon1) [reply build/fake_azure.py:88]: I'm thinking of just renaming it and not splitting it into multiple Bottle applications, because the mock services at this point are so similar and use identical helper functions.

File: src/libmongoc/src/mongoc/mongoc-error.h
Comment (vector-of-bool) [inline src/libmongoc/src/mongoc/mongoc-error.h:142]: @kevinAlbs Do we want to keep the old names for these constants present? Have they been released for long enough to be relied on?

**Review by vector-of-bool — APPROVED** [review]
LGTM, pending one comment.

File: src/libmongoc/src/mongoc/mongoc-error.h
Comment (kevinAlbs) [reply src/libmongoc/src/mongoc/mongoc-error.h:142]: No need. The constants were introduced in CDRIVER-4454 and have not been released yet.

**Review by kevinAlbs — APPROVED** [review]
Nicely done. Before merging this PR, please run an Evergreen patch build with the two new tasks. The two new tasks are not run by default.
That can be done with:
```
evergreen patch --project=mongo-c-driver --description="Test GCP $(git log --oneline HEAD~1..HEAD)" \
    --yes --finalize \
    -v testgcpkms-variant -t testgcpkms-task \
    -v testgcpkms-variant -t testgcpkms-fail-task
```

File: build/evergreen_config_lib/testgcpkms.py
Comment (galon1) [reply build/evergreen_config_lib/testgcpkms.py:80]: I made it debian 10 because I couldn't download libmongocrypt using debian 11 on my local computer since the package wasn't available yet. Is it fine though for the GCP virtual machine to be on debian 11?

Comment (kevinAlbs) [reply build/evergreen_config_lib/testgcpkms.py:80]: The GCP virtual machine is already using Debian 11. Suggested change is to have the OS in Evergreen match the GCP OS. libmongocrypt is built from source in Evergreen, so I expect it should be OK.

File: .evergreen/mongo-c-driver.spec
Comment (rcsanchez97) [reply .evergreen/mongo-c-driver.spec:46]: @galon1 I concur with reverting the changes to this file.  It is only updated when the downstream maintainer publishes a new version.  If you really need to modify the spec file (i.e., to keep the `rpm-package-build` task green), then you can make the changes you need and regenerate `.evergreen/spec.patch`.  Don't forget to first apply `spec.patch` to preserve the existing changes, and when you regenerate make sure that it will apply with a patch strip level of 0.

Comment (galon1) [reply .evergreen/mongo-c-driver.spec]: Just to confirm if we want to upgrade the dependency on libmongocrypt to a newer version like 1.6.0, this file doesn't need to be changed. This file is only changed when a external third party publishes a new version and we want to stay up to date with their version?

**Comment by rcsanchez97** [comment]
Essentially, that is the case.  However, the Linux distro packaging
tasks are a little more complicated than that.  In this particular
instance, you don't need to do anything.  The mongo-c-driver spec calls
for `(mongocrypt) >= 1.5.2`.  However, the current target environment of
the `.evergreen/build_snapshot_rpm.sh` script is Fedora 37.  The current
version of
[libmongocrypt](https://src.fedoraproject.org/rpms/libmongocrypt/) in
Fedora 37 is 1.6.0.  So, your changes will "just work" without any
modifications to the way the RPM builds.

File: .evergreen/mongo-c-driver.spec
Comment (rcsanchez97) [reply .evergreen/mongo-c-driver.spec]: Essentially, that is the case.  However, the Linux distro packaging tasks are a little more complicated than that.  In this particular instance, you don't need to do anything.  The mongo-c-driver spec calls for `(mongocrypt) >= 1.5.2`.  However, the current target environment of the `.evergreen/build_snapshot_rpm.sh` script is Fedora 37.  The current
version of [libmongocrypt](https://src.fedoraproject.org/rpms/libmongocrypt/) in Fedora 37 is 1.6.0.  So, your changes will "just work" without any modifications to the way the RPM builds.

Comment (galon1) [reply .evergreen/mongo-c-driver.spec]: That makes sense. Thank you! I reverted my changes

**Pull request merged**
